modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/snapshot/MajorityPeerSelectionStrategy.scala

```diff
@@ -0,0 +1,53 @@
+package org.tessellation.sdk.infrastructure.snapshot
+
+import cats.data.NonEmptyList
+import cats.effect.kernel.Async
+import cats.effect.std.Random
+import cats.syntax.all._
+
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.schema.node.NodeState.Ready
+import org.tessellation.schema.peer.Peer
+import org.tessellation.schema.peer.Peer.toP2PContext
+import org.tessellation.sdk.domain.cluster.storage.ClusterStorage
+import org.tessellation.sdk.domain.snapshot.peerSelection.{NoPeers, PeerSelectionStrategy}
+import org.tessellation.sdk.http.p2p.clients.L0GlobalSnapshotClient
+
+class MajorityPeerSelectionStrategy[F[_]: Async: KryoSerializer: Random](
+  val storage: ClusterStorage[F],
+  val snapshotClient: L0GlobalSnapshotClient[F]
+) extends PeerSelectionStrategy[F] {
+  def select: F[Peer] = for {
+    readyPeers <- storage.getResponsivePeers
+      .map(_.filter(_.state === Ready))
+    peersToQuery <- getPeerSublist(readyPeers).flatMap { peers =>
+      Async[F].fromOption(peers.toNel, NoPeers)
+    }
+    peerCandidates <- filterPeerList(peersToQuery)
+    selectedPeer <- Random[F].elementOf(peerCandidates.toList)
+  } yield selectedPeer
```

Comment:
in general, you just create next variables based on the previous ones, so for comprehension is imo less readable. please consider this:

```scala
storage.getResponsivePeers
  .map(_.filter(_.state === Ready))
  .flatMap(getPeerSublist)
  .flatMap(peers => ApplicativeError[F].fromOption(peers.toNel, NoPeers))
  .flatMap(filterPeerList)
  .flatMap(Random[F].elementOf(_))
```

Reply:
Changed.